# Sobre o notebook

A "hora da prática" é uma atividade do curso de Python para Data Science da Alura que contém diversos desafios envolvendo os aprendizados de cada aula.

O notebook "hora da pratica" é um notebook construído com espaço para as células dos desafios propostos no curso, com ele você tem espaço para construir e executar suas soluções

## Aula 1 - Introdução ao Python

### Questão 1

Imprima a frase `Escola de Dados da Alura!`

In [ ]:
print('Escola de Dados da Alura!')

Escola de Dados da Alura!


### Questão 2
Imprima na tela seu nome e seu sobrenome seguindo a estrutura abaixo:
```
Nome: [seu nome]
Sobrenome: [seu sobrenome]
```


In [ ]:
print('Nome: Lucas')
print('Sobrenome: Coutinho')

Nome: Lucas
Sobrenome: Coutinho


### Questão 3
Imprima o seu primeiro nome letra a letra. Por exemplo, nome é Mirla, então eu obtenho a seguinte saída:
```
M
I
R
L
A
```


In [ ]:
print('L')
print('U')
print('C')
print('A')
print('S')

L
U
C
A
S


### Questão 4
Imprima o dia do seu nascimento em formato `dia mês ano`. Os valores de dia e ano não podem estar entre aspas. O formato deve estar como no exemplo, supondo uma data de aniversário dia 28 de fevereiro de 2003:
```
28 fevereiro 2003
```

In [ ]:
print('30 setembro 1990')

30 setembro 1990


### Questão 5
Imprima em um único print o atual ano que você está fazendo esse curso, o valor do ano deve ser um dado numérico. A saída do print deve ser a seguinte:
```
Ano atual: [ano]
```


In [ ]:
print('Ano atual:2023')

Ano atual:2023


## Aula 2 - Manipulando dados no Python

### Coleta e amostragem de dados

#### Questão 1
Crie um programa que solicite à pessoa usuária digitar seu nome, e imprima “Olá, [nome]!”.

In [ ]:
nome = input('Insira seu nome: ')
print(nome)


Insira seu nome: Lucas
Lucas


#### Questão 2
Crie um programa que solicite à pessoa usuária digitar seu nome e idade, e imprima “Olá, [nome], você tem [idade] anos.”.

In [ ]:
nome = input('Qual seu nome:')
idade = input('Quantos anos você tem?')
print(f'Olá {nome}, você tem {idade} anos')

Qual seu nome:Lucas
Quantos anos você tem?33
Olá Lucas, você tem 33 anos


#### Questão 3
Crie um programa que solicite à pessoa usuária digitar seu nome, idade e altura em metros, e imprima “Olá, [nome], você tem [idade] anos e mede [altura] metros!”.

In [ ]:
nome = input('Qual seu nome:')
idade = input('Quantos anos você tem?')
altura = input('Qual é a sua altura?')
print(f'Olá {nome}, você tem {idade} anos e mede {altura} metros')

Qual seu nome:Lucas
Quantos anos você tem?33
Qual é a sua altura?1,83
Olá Lucas, você tem 33 anos e mede 1,83 metros


### Calculadora com operadores

#### Questão 1

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a soma dos dois valores.

In [ ]:
n1 = input('pensa em um numero: ')
n2 = input (' agora pensa em outro: ')
soma = int(n1)+int(n2)
print(f' a soma de {n1} + {n2} é {soma}')



pensa em um numero5
 agora pensa em outro9
 a soma de 5+9 é 14


#### Questão 2
Crie um programa que solicite três valores numéricos à pessoa usuária e imprima a soma dos três valores.

In [ ]:
n1 = input('pensa em um numero: ')
n2 = input('agora pensa em outro: ')
n3 = input('agora mais um: ')
soma = int(n1) + int(n2) + int(n3)
print(f'a soma de {n1} + {n2} + {n3} é {soma}')


pensa em um numero: 7
agora pensa em outro: 4
agora mais um: 8
a soma de 7 + 4 + 8 é 19


#### Questão 3
Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a subtração do primeiro pelo o segundo valor.

In [ ]:
n1 = input('pensa em um numero')
n2 = input (' agora pensa em outro')
sub = int(n1)-int(n2)
print(f'{n1} - {n2} é {sub}')

pensa em um numero8
 agora pensa em outro4
8 - 4 é 4


#### Questão 4

Crie um programa que solicite dois valores numéricos à pessoa usuária e imprima a multiplicação dos dois valores.


In [2]:
n1 = int(input('Pensa num número: '))
n2 = int(input('Pensa em outro número'))
print(f'{n1} multiplicado por {n2} é {n1*n2}.')

Pensa num número: 9
Pensa em outro número45
9 multiplicado por 45 é 405.


#### Questão 5

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e realize a divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.


In [7]:
n1 = int(input('Pensa num número: '))
n2 = int(input('Pensa em outro número para ser o denominador da divisão: '))
if n2 == 0:
  while n2 == 0:
    n2 = int(input('O denominador não pode ser zero. Digite outro número diferente de zero: '))
print(f'{n1} dividido por {n2} é {n1/n2}.')

Pensa num número: 9846513246987465
Pensa em outro número para ser o denominador da divisão: 0
O denominador não pode ser zero. Digite outro número diferente de zero: 0
O denominador não pode ser zero. Digite outro número diferente de zero: 0
O denominador não pode ser zero. Digite outro número diferente de zero: 0
O denominador não pode ser zero. Digite outro número diferente de zero: 7984651
9846513246987465 dividido por 7984651 é 1233180166.1697505.


#### Questão 6

Crie um programa que solicite dois valores numéricos, um operador e uma potência, e realize a exponenciação entre esses dois valores.


#### Questão 7
Crie um programa que solicite dois valores numéricos, um numerador e um denominador e realize a divisão inteira entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

#### Questão 8

Crie um programa que solicite dois valores numéricos, um numerador e um denominador, e retorne o resto da divisão entre os dois valores. Deixe claro que o valor do denominador não pode ser `0`.

#### Questão 9

Crie um código que solicita 3 notas de um estudante e imprima a média das notas.

#### Questão 10

Crie um código que calcule e imprima a média ponderada dos números 5, 12, 20 e 15 com pesos respectivamente iguais a 1, 2, 3 e 4.

### Editando textos

#### Questão 1

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase na tela.

#### Questão 2
Crie um código que solicite uma frase e depois imprima a frase na tela.

#### Questão 3

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras maiúsculas.

#### Questão 4

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase digitada mas com todas as letras minúsculas.

#### Questão 5

Crie uma variável chamada “`frase`” e atribua a ela uma string de sua escolha. Em seguida, imprima a frase sem espaços em branco no início e no fim.

#### Questão 6

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim.

#### Questão 7


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase sem espaços em branco no início e no fim e em letras minúsculas.

#### Questão 8

Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`e`” trocadas pela letra “`f`”.

#### Questão 9


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as vogais “`a`” trocadas pela caractere  “`@`”.

#### Questão 10


Crie um código que solicite uma frase à pessoa usuária e imprima a mesma frase com todas as consoantes “`s`” trocadas pelo caractere  “`$`”.

## Aula 3 - Estruturas condicionais

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça à pessoa usuária para fornecer dois números e exibir o número maior.

In [ ]:
n1 = input('Escreva um numero: ')
n2 = input('Escreva outro numero: ')
if n1 > n2:
  print(f'O primeiro numero é maior que o segundo')
else:
  print('O segundo é maior que o primeiro')

Escreva um numero: 1
escreva outro numero: 5
o segundo é maior que o primeiro


#### Questão 2
Escreva um programa que solicite o percentual de crescimento de produção de uma empresa e informe se houve um crescimento (porcentagem positiva) ou decrescimento (porcentagem negativa).

In [ ]:
growth = input('Quantos porcento cesceu as produções?: ')
if int(growth) <0:
  print('Houve um decresimo na produção de {}%.'.format(growth))
else:
  print('Houve um crescimento na produção de {}%.'.format(growth))


Quantos porcento cesceu as produções?: -9
Houve um decresimo na produção de -9%.


#### Questão 3
Escreva um programa que determine se uma letra fornecida pela pessoa usuária é uma vogal ou consoante.

In [ ]:
letra = input("Digite uma letra: ")
lista = 'a, e, i, o, u'
if letra in lista:
  print('A letra é uma vogal')
else:
  print('A letra é uma consosante')

Digite uma letra: o
A letra é uma vogal


#### Questão 4
Escreva um programa que leia valores médios de preços de um modelo de carro por 3 anos consecutivos e exiba o valor mais alto e mais baixo entre esses três anos.

#### Questão 5
Escreva um programa que pergunte sobre o preço de três produtos e indique qual é o produto mais barato para comprar.

In [12]:
produto1 = float(input('Quanto custa o produto 1? '))
produto2 = float(input('Quanto custa o produto 2? '))
produto3 = float(input('Quanto custa o produto 3? '))
lista_precos = []
lista_precos.extend([produto1, produto2, produto3])
lista_precos.sort()
print(f'O preço mais barato é o {lista_precos[0]}.')

Quanto custa o produto 1? 8.44
Quanto custa o produto 2? 78.11
Quanto custa o produto 3? 1.77
O preço mais barato é o 1.77.


#### Questão 6
Escreva um programa que leia três números e os exiba em ordem decrescente.

#### Questão 7
Escreva um programa que pergunte em qual turno a pessoa usuária estuda ("manhã", "tarde" ou "noite") e exiba a mensagem "Bom Dia!", "Boa Tarde!", "Boa Noite!", ou "Valor Inválido!", conforme o caso.

#### Questão 8
Escreva um programa que peça um número inteiro à pessoa usuária e determine se ele é par ou ímpar. Dica: Você pode utilizar o operador módulo `%`.

#### Questão 9
Escreva um programa que peça um número à pessoa usuária e informe se ele é inteiro ou decimal.

### Momento dos projetos

#### Questão 10
Um programa deve ser escrito para ler dois números e, em seguida, perguntar à pessoa usuária qual operação ele deseja realizar. O resultado da operação deve incluir informações sobre o número - se é par ou ímpar, positivo ou negativo e inteiro ou decimal.

#### Questão 11
Escreva um programa que peça à pessoa usuária três números que representam os lados de um triângulo. O programa deve informar se os valores podem ser utilizados para formar um triângulo e, caso afirmativo, se ele é equilátero, isósceles ou escaleno. Tenha em mente algumas dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;


#### Questão 12
Um estabelecimento está vendendo combustíveis com descontos variados. Para o etanol, se a quantidade comprada for até 15 litros, o desconto será de 2% por litro. Caso contrário, será de 4% por litro. Para o diesel, se a quantidade comprada for até 15 litros, o desconto será de 3% por litro. Caso contrário, será de 5% por litro. O preço do litro de diesel é R\$ 2,00 e o preço do litro de etanol é R\$ 1,70. Escreva um programa que leia a quantidade de litros vendidos e o tipo de combustível (E para etanol e D para diesel) e calcule o valor a ser pago pelo cliente. Tenha em mente algumas dicas:

- O do valor do desconto será a multiplicação entre preço do litro, quantidade de litros e o valor do desconto.
- O valor a ser pago por um cliente será o resultado da multiplicação do preço do litro pela quantidade de litros menos o valor de desconto resultante do cálculo.


#### Questão 13
Em uma empresa de venda de imóveis você precisa criar um código que analise os dados de vendas anuais para ajudar a diretoria na tomada de decisão. O código precisa coletar os dados de quantidade de venda durante os anos de 2022 e 2023 e fazer um cálculo de variação percentual. A partir do valor da variação, deve ser enviada às seguintes sugestões:
- Para variação acima de 20%: bonificação para o time de vendas.
- Para variação entre 2% e 20%: pequena bonificação para time de vendas.
- Para variação entre 2% e -10%: planejamento de políticas de incentivo às vendas.
- Para bonificações abaixo de -10%: corte de gastos.


## Aula 4 - Estruturas de repetição

### Aquecendo na programação

#### Questão 1
Escreva um programa que peça dois números inteiros e imprima todos os números inteiros entre eles.

In [ ]:
inicio = int(input('De qual valor vamos começar? '))
fim = int(input('De qual valor vamos terminar? '))
while inicio <= fim:
  print(inicio)
  inicio += 1

De qual valor vamos começar? 9
De qual valor vamos terminar? 22
9
10
11
12
13
14
15
16
17
18
19
20
21
22


#### Questão 2
Escreva um programa para calcular quantos dias levará para a colônia de uma bactéria A ultrapassar ou igualar a colônia de uma bactéria B, com base nas taxas de crescimento de 3% e 1,5% respectivamente. Considere que a colônia A inicia com 4 elementos e a colônia B com 10 elementos.

In [15]:
A = 4
B = 10
D = 0
while A <= B:
  print("Bactéria A = {:.2f}, Bactéria B = {:.2f}".format(A,B))
  A *= 1.03
  B *= 1.015
  D += 1
print(f'{D} dias')

Bactéria A = 4.00, Bactéria B = 10.00
Bactéria A = 4.12, Bactéria B = 10.15
Bactéria A = 4.24, Bactéria B = 10.30
Bactéria A = 4.37, Bactéria B = 10.46
Bactéria A = 4.50, Bactéria B = 10.61
Bactéria A = 4.64, Bactéria B = 10.77
Bactéria A = 4.78, Bactéria B = 10.93
Bactéria A = 4.92, Bactéria B = 11.10
Bactéria A = 5.07, Bactéria B = 11.26
Bactéria A = 5.22, Bactéria B = 11.43
Bactéria A = 5.38, Bactéria B = 11.61
Bactéria A = 5.54, Bactéria B = 11.78
Bactéria A = 5.70, Bactéria B = 11.96
Bactéria A = 5.87, Bactéria B = 12.14
Bactéria A = 6.05, Bactéria B = 12.32
Bactéria A = 6.23, Bactéria B = 12.50
Bactéria A = 6.42, Bactéria B = 12.69
Bactéria A = 6.61, Bactéria B = 12.88
Bactéria A = 6.81, Bactéria B = 13.07
Bactéria A = 7.01, Bactéria B = 13.27
Bactéria A = 7.22, Bactéria B = 13.47
Bactéria A = 7.44, Bactéria B = 13.67
Bactéria A = 7.66, Bactéria B = 13.88
Bactéria A = 7.89, Bactéria B = 14.08
Bactéria A = 8.13, Bactéria B = 14.30
Bactéria A = 8.38, Bactéria B = 14.51
Bactéria A =

#### Questão 3
Para tratar uma quantidade de 15 dados de avaliações de pessoas usuárias de um serviço da empresa, precisamos verificar se as notas são válidas. Então, escreva um programa que vai receber a nota de 0 a 5 de todos os dados e verificar se é um valor válido. Caso seja inserido uma nota acima de 5 ou abaixo de 0, repita até que a pessoa usuária insira um valor válido.

In [16]:
i = 1
nota = 0
lista = []
for i in range(1,16):
  nota = int(input('escreva a nota do usuário: '))
  if nota >= 0 and nota <= 5:
    lista. append(nota)
    continue
  else:
    while nota < 0 or nota > 5:
      print('Nota fora dos paramentros. Insira um valor entre 0 e 5.')
      nota = int(input('escreva a nota do usuário: '))
  lista. append(nota)
lista


escreva a nota do usuário: 1
escreva a nota do usuário: 2
escreva a nota do usuário: 5
escreva a nota do usuário: 4
escreva a nota do usuário: 3
escreva a nota do usuário: 6
Nota fora dos paramentros. Insira um valor entre 0 e 5.
escreva a nota do usuário: 0
escreva a nota do usuário: 1
escreva a nota do usuário: 8
Nota fora dos paramentros. Insira um valor entre 0 e 5.
escreva a nota do usuário: 1
escreva a nota do usuário: 2
escreva a nota do usuário: 3
escreva a nota do usuário: 6
Nota fora dos paramentros. Insira um valor entre 0 e 5.
escreva a nota do usuário: 4
escreva a nota do usuário: 5
escreva a nota do usuário: 2
escreva a nota do usuário: 3
escreva a nota do usuário: 1


[1, 2, 5, 4, 3, 0, 1, 1, 2, 3, 4, 5, 2, 3, 1]

#### Questão 4
Desenvolva um programa que leia um conjunto indeterminado de temperaturas em Celsius e faça uma análise. Portanto, escreva um programa que leia temperaturas e informe a média delas. A leitura deve ser encerrada ao ser enviado o valor -273°C.

#### Questão 5
Escreva um programa que calcule o fatorial de um número inteiro fornecido pela pessoa usuária. O fatorial de um número inteiro é a multiplicação desse número por todos os seus antecessores até o número 1. Por exemplo, o fatorial de 5 é 5 x 4 x 3 x 2 x 1 = 120.

### Momento dos projetos

#### Questão 6
Escreva um programa que gere a tabuada de um número inteiro de 1 a 10, de acordo com a escolha da pessoa usuária. Como exemplo, para o número 2, a tabuada deve ser mostrada no seguinte formato:
```
Tabuada do 2:
2 x 1 = 2
2 x 2 = 4
[...]
2 x 10 = 20
```

In [ ]:
contador = 1
N = int(input('Qual número você quer? '))
for contador in range(1,11):
  print(N * contador)


Qual número você quer? 4
4
8
12
16
20
24
28
32
36
40


#### Questão 7
Os números primos possuem várias aplicações dentro da Ciência de Dados, por exemplo, na criptografia e segurança. Um número primo é aquele que é divisível apenas por um e por ele mesmo. Faça um programa que peça um número inteiro e determine se ele é ou não um número primo.

#### Questão 8
Vamos entender a distribuição de idades de pensionistas de uma empresa de previdência. Escreva um programa que leia as idades de uma quantidade não informada de clientes e mostre a distribuição em intervalos de [0-25], [26-50], [51-75] e [76-100]. Encerre a entrada de dados com um número negativo.

#### Questão 9
Em uma eleição para gerência em uma empresa com 20 funcionários, existem quatro candidatos. Escreva um programa que calcule o vencedor da eleição. A votação ocorreu da seguinte maneira:
- Cada funcionário votou em um dos quatro candidatos (representados pelos números 1, 2, 3 e 4).
- Também foram contabilizados os votos nulos (representado pelo número 5) e os votos em branco (representado pelo número 6).

Ao final da votação, o programa deve exibir o total de votos para cada candidato, o número de votos nulos e o número de votos em branco. Além disso, deve calcular e exibir a porcentagem de votos nulos em relação ao total de votos e a porcentagem de votos em branco em relação ao total de votos.

## Aula 5 - Estruturas de dados

### Aquecendo na programação

#### Questão 1
Faça um programa que tenha a seguinte lista contendo os valores de gastos de uma empresa de papel `[2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]`. Com esses valores, faça um programa que calcule a média de gastos. Dica, use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).

In [18]:
Gastos_empresa = [2172.54, 3701.35, 3518.09, 3456.61, 3249.38, 2840.82, 3891.45, 3075.26, 2317.64, 3219.08]
media = sum(Gastos_empresa) / len(Gastos_empresa)
media

3144.222

#### Questão 2
 Com os mesmos dados da questão anterior, defina quantas compras foram acima de 3000 reais e calcule a porcentagem quanto ao total de compras.

#### Questão 3
Faça um código que colete em uma lista 5 números inteiros quaisquer e imprima a lista. Exemplo: `[1,4,7,2,4]`.

#### Questão 4
Colete novamente 5 inteiros e imprima a lista em ordem inversa à enviada.

#### Questão 5
Faça um programa que, ao inserir um número qualquer, criará uma lista contendo todos os números primos entre 1 e o número digitado.

#### Questão 6
Escreva um programa que peça uma data informando o dia, mês e ano e determine se ela é uma data válida para uma análise.

### Momento dos projetos

#### Questão 7
Para um estudo envolvendo o nível de multiplicação de bactérias em uma colônia, foi coletado o número de bactérias multiplicadas por dia, com base em um processo de duplicação diária e pode ser observado a seguir: `[1.2, 2.1, 3.3, 5.0, 7.8, 11.3, 16.6, 25.1, 37.8, 56.9]`. Tendo esses valores, faça um código que gere uma lista contendo o percentual de crescimento de bactérias por dia, comparando o número de bactérias em cada dia com o número de bactérias do dia anterior. Dica: para calcular o percentual de crescimento usamos a seguinte equação: `100 * (amostra_atual - amostra_passada) / (amostra_passada)`.

#### Questão 8
Para uma seleção de produtos alimentícios, precisamos separar o conjunto de IDs dados por números inteiros, sabendo que os produtos com ID par são doces e os com ID ímpar são amargos. Monte um código que colete 10 IDs. Depois, calcule e mostre a quantidade de produtos doces e amargos.

#### Questão 9
Desenvolva um programa que informa a nota de um aluno de acordo com suas respostas. Ele deve pedir a resposta de um aluno para cada questão e é preciso verificar se a resposta foi igual ao gabarito. Cada questão vale um ponto e existem alternativas:  A, B, C ou D.
```
Gabarito da Prova:
01 - D
02 - A
03 - C
04 - B
05 - A
06 - D
07 - C
08 - C
09 - A
10 - B
```

#### Questão 10
Um instituto de meteorologia deseja fazer um estudo de temperatura média de cada mês do ano. Para isso, você precisa fazer um código que colete e armazene essas temperaturas médias em uma lista. Depois, calcule a média anual das temperaturas e mostre todas as temperaturas acima da média anual, e em que mês elas ocorreram, mostrando os meses por extenso: Janeiro, Fevereiro, etc.

#### Questão 11
Uma empresa de e-commerce está interessada em analisar as vendas dos seus produtos. Os dados das vendas foram armazenados em um dicionário:
```
{'Produto A': 300, 'Produto B': 80, 'Produto C': 60,
 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}
```
Escreva um código que calcule o total de vendas e o produto mais vendido.

In [51]:
vendas = {'Produto A': 300, 'Produto B': 80, 'Produto C': 60,
 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}
soma = sum(vendas.values())
for i in sorted(vendas, key = vendas.get):
    print(i, vendas[i])


Produto F 30
Produto C 60
Produto B 80
Produto D 200
Produto E 250
Produto A 300


In [52]:
# Dicionário de vendas
dados_vendas = {'Produto A': 300, 'Produto B': 80, 'Produto C': 60, 'Produto D': 200, 'Produto E': 250, 'Produto F': 30}

# Inicializamos as variáveis
total_vendas = 0 # Irá somar todos as vendas
produto_mais_vendido = '' # Irá armazenar o nome do produto mais vendido
unidades_produto_mais_vendido = 0 # Irá armazenar a maior quantidade vendas

# Percorremos os valores de chaves e elementos do dicionário
for produto in dados_vendas.keys():
  # Somamos o total de vendas
  total_vendas += dados_vendas[produto]
  # Verificamos se valor de venda atual desing (dados_vendas[produto]) é maior que o valor armazenado em unidades_produto_mais_vendido
  # Cada vez que dados_vendas[produto] superar o valor em unidades_produto_mais_vendido,
  # a variável unidades_produto_mais_vendido vai ser igual à dados_vendas[produto], atribuindo um novo valor
  # De forma similar, produto_mais_vendido também é substituído pelo produto atual
  if dados_vendas[produto] > unidades_produto_mais_vendido:
    unidades_produto_mais_vendido = dados_vendas[produto]
    produto_mais_vendido = produto
# Resultados
print(f'Total de vendas é {total_vendas}')
print(f'{produto_mais_vendido} é o mais vendido')


Total de vendas é 920
Produto A é o mais vendido


#### Questão 12
Uma pesquisa de mercado foi feita para decidir qual design de uma marca infantil mais agrada crianças. A pesquisa foi feita e o votos computados podem ser observados abaixo:
```
'''
Tabela de votos da marca
Design 1 - 1334 votos
Design 2 - 982 votos
Design 3 - 1751 votos
Design 4 - 210 votos
Design 5 - 1811 votos
'''
```
Adapte os dados fornecidos a você para uma estrutura de dicionário e a partir dele, informe o design vencedor e a porcentagem de votos recebidos.


#### Questão 13
Os funcionários de um setor da empresa que você trabalha vão receber um abono correspondente a 10% do seu salário devido ao ótimo desempenho do time. O setor financeiro solicitou sua ajuda para a verificação das consequências financeiras que esse abono irá gerar nos recursos. Assim, foi encaminhada para você uma lista com os salários que receberão o abono: `[1172, 1644, 2617, 5130, 5532, 6341, 6650, 7238, 7685, 7782, 7903]`. O abono de cada funcionário não pode ser inferior a 200. Em código, transforme cada um dos salários em chaves de um dicionário e o abono de cada salário no elemento. Depois, informe o total de gastos com o abono, quantos funcionários receberam o abono mínimo e qual o maior valor de abono fornecido.


#### Questão 14
Uma equipe de cientistas de dados está estudando a diversidade biológica em uma floresta. A equipe fez a coleta de informações sobre o número de espécies de plantas e animais em cada área da floresta e armazenaram essas informações em um dicionário. Nele, a chave descreve a área dos dados e os valores nas listas correspondem às espécies de plantas e animais nas áreas, respectivamente.
```
{'Área Norte': [2819, 7236],
 'Área Leste': [1440, 9492],
 'Área Sul': [5969, 7496],
 'Área Oeste': [14446, 49688],
 'Área Centro': [22558, 45148]}
```
Escreva um código para calcular a média de espécies por área e identificar a área com a maior diversidade biológica. Dica: use as funções built-in [`sum()`](https://docs.python.org/3/library/functions.html#sum) e [`len()`](https://docs.python.org/3/library/functions.html#len).




#### Questão 15
O setor de RH da sua empresa te pediu uma ajuda para analisar as idades dos funcionários de 4 setores da empresa. Para isso, ele te forneceu os seguintes dados:
```
{'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}
```
Sabendo que cada setor tem 10 funcionários, construa um código que calcule a média de idade de cada setor, a idade média geral entre todos os setores e quantas pessoas estão acima da idade média geral.


In [ ]:
empresa = {'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}

for chave in empresa.keys():
  soma = sum(empresa[chave])
  media = soma / len(empresa[chave])
  print(f'{chave}, média: {media} anos.')

Setor A, média: 39.9 anos.
Setor B, média: 42.3 anos.
Setor C, média: 35.8 anos.
Setor D, média: 36.5 anos.


In [ ]:
empresa = {'Setor A': [22, 26, 30, 30, 35, 38, 40, 56, 57, 65],
 'Setor B': [22, 24, 26, 33, 41, 49, 50, 54, 60, 64],
 'Setor C': [23, 26, 26, 29, 34, 35, 36, 41, 52, 56],
 'Setor D': [19, 20, 25, 27, 34, 39, 42, 44, 50, 65]}

for x in empresa.keys():
  print(empresa[x])

[22, 26, 30, 30, 35, 38, 40, 56, 57, 65]
[22, 24, 26, 33, 41, 49, 50, 54, 60, 64]
[23, 26, 26, 29, 34, 35, 36, 41, 52, 56]
[19, 20, 25, 27, 34, 39, 42, 44, 50, 65]
